spark/core.py

```diff
@@ -303,19 +316,45 @@ def fill_arc(self, *args):
 
     def stroke_arc(self, *args):
         self.canvas.stroke_arc(*args)
-    
-    def fill_text(self, *args):
-        self.canvas.font = "{px}px sans-serif".format(px = args[4])
-        self.canvas.fill_text(args[0:3])
-        self.canvas.font = "12px sans-serif"
 
-    def stroke_text(self, *args):
-        self.canvas.font = "{px}px sans-serif".format(px = args[4])
-        self.canvas.stroke_text(args[0:3])
-        self.canvas.font = "12px sans-serif"
+    def text_size(self, *args):
+        if len(args) != 1:
+            raise TypeError(f"text_size expected 1 argument, got {len(args)}")
+        
+        size = args[0]
+        if type(size) == type(str()):
+            if size.endswith('px'):
+                size = size[:-2] # Remove 'px' from end.
+            else:
+                raise TypeError(f"text_size expects a number or a string with a number followed by 'px', got {repr(args[0])}")
+
+        try:
+            size = float(size)
+        except ValueError:
+            raise TypeError(f"text_size expects a number or a string with a number followed by 'px', got {repr(args[0])}")
+
+        self.font_settings['size'] = size
+        self.canvas.font = f"{self.font_settings['size']}px {self.font_settings['font']}"
 
     def text_align(self, *args):
-        self.canvas.text_align(*args)
+        if len(args) != 1:
+            raise TypeError(f"text_size expected 1 argument, got {len(args)}")
+
+        if args[0] not in ['left', 'right', 'center']:
+            raise TypeError(f"text_align expects a string of 'left', 'right', or 'center', got {args[0]}")
+
+        self.canvas.text_align = args[0]
+
+    def text(self, *args):
+        if len(args) != 3:
+            raise TypeError(f"text expected 3 arguments, got {len(args)}")
```

Comment:
Since I always get the parameter order mixed up, I might recommend adding the expected params into here:
```suggestion
            raise TypeError(f"text expected 3 arguments (message, x, y), got {len(args)}")
```